controllers/users.js

```diff
@@ -0,0 +1,65 @@
+const userModel = require('../db/user.model');
+
+class UsersController {
+    static addUser(req, res) {
+        const { username, password } = req.body; 
+        if(!username || !password) return res.status(400);
+        if(userModel.find({ username }).length > 0) {
+            return res.status(409).json({ error: 'Username already exists' });
+        }
+        try {
+            const newUser = userModel.build({ 
+                id: (Math.ceil(Math.random() * Math.pow(10, 6))).toString(), 
+                username, 
+                password 
+            });
+            newUser.save();
+            return res.sendStatus(200);
+        } catch (error) {
+            return res.sendStatus(500);
+        }
+    }
+    static getUser(req, res) {
```

Comment:
No need to reuse this function like this, it's much more readable, that you have 2 different functions.

getUsers and getUser, try to write functions, that do one thing, without having some hidden logic inside.

Reply:
Got it. But the route should still be the same both for all posts and an individual post, right?

Reply:
No, it should be different, for 1 user it will be 

```javascript
router.get('/:id', userCtrl.getUser);
router.get('/', userCtrl.getUsers);
```

Reply:
No, it should be different

```javascript
router.get(':id', user)
router.get('/', users)
```